rust/feature-flags/src/flag_definitions.rs

```diff
@@ -298,4 +404,948 @@ mod tests {
+            deleted: false,
+            active: true,
+            ensure_experience_continuity: false,
+        };
+
+        let flag2 = FeatureFlagRow {
+            id: random_id_2,
+            team_id: team.id,
+            name: Some("Test Flag 2".to_string()),
+            key: "test_flag_2".to_string(),
+            filters: serde_json::json!({"groups": [{"properties": [], "rollout_percentage": 100}]}),
+            deleted: false,
+            active: true,
+            ensure_experience_continuity: false,
+        };
+
+        // Insert multiple flags for the team
+        insert_flag_for_team_in_pg(client.clone(), team.id, Some(flag1))
+            .await
+            .expect("Failed to insert flags");
+
+        insert_flag_for_team_in_pg(client.clone(), team.id, Some(flag2))
+            .await
+            .expect("Failed to insert flags");
+
+        let flags_from_pg = FeatureFlagList::from_pg(client.clone(), team.id)
+            .await
+            .expect("Failed to fetch flags from pg");
+
+        assert_eq!(flags_from_pg.flags.len(), 2);
+        for flag in &flags_from_pg.flags {
+            assert_eq!(flag.team_id, team.id);
+        }
+    }
+}
+
+#[cfg(test)]
+mod additional_tests {
+    use std::time::Instant;
+
+    use crate::{
+        flag_definitions,
+        test_utils::{
+            insert_flag_for_team_in_pg, insert_flags_for_team_in_redis, insert_new_team_in_pg,
+            setup_pg_client, setup_redis_client,
+        },
+    };
+
+    use super::*;
+    use serde_json::json;
+    use tokio::task;
+
+    #[test]
+    fn test_operator_type_deserialization() {
+        let operators = vec![
+            ("exact", OperatorType::Exact),
+            ("is_not", OperatorType::IsNot),
+            ("icontains", OperatorType::Icontains),
+            ("not_icontains", OperatorType::NotIcontains),
+            ("regex", OperatorType::Regex),
+            ("not_regex", OperatorType::NotRegex),
+            ("gt", OperatorType::Gt),
+            ("lt", OperatorType::Lt),
+            ("gte", OperatorType::Gte),
+            ("lte", OperatorType::Lte),
+            ("is_set", OperatorType::IsSet),
+            ("is_not_set", OperatorType::IsNotSet),
+            ("is_date_exact", OperatorType::IsDateExact),
+            ("is_date_after", OperatorType::IsDateAfter),
+            ("is_date_before", OperatorType::IsDateBefore),
+        ];
+
+        for (op_str, op_type) in operators {
+            let json = format!(
+                r#"{{
+            "key": "test_key",
+            "value": "test_value",
+            "operator": "{}",
+            "type": "person"
+        }}"#,
+                op_str
+            );
+            let deserialized: PropertyFilter = serde_json::from_str(&json).unwrap();
+            assert_eq!(deserialized.operator, Some(op_type));
+        }
+    }
+
+    #[tokio::test]
+    async fn test_multivariate_flag_parsing() {
```

Comment:
We should add another test for multivariate flags with payloads attached to each variant here.

Reply:
done